src/components/dev-hub/feedback/feedback-container.js

```diff
@@ -11,21 +18,134 @@ const feedbackItems = graphql`
     }
 `;
 
-const FeedbackContainer = () => {
-    const data = useStaticQuery(feedbackItems);
+const StyledButtonContainer = styled('div')`
+    display: flex;
+    justify-content: center;
+    margin-bottom: 4px;
+`;
+
+const StyledForm = styled('form')`
+    padding: 0 ${size.medium};
+
+    button,
+    label,
+    textarea {
+        margin: ${size.default} 0;
+    }
+
+    input {
+        margin-bottom: ${size.default};
+    }
+`;
+
+const StyledDescription = styled(P)`
+    display: inline-block;
+    font-family: Akzidenz;
+    font-size: ${fontSize.small};
+    line-height: ${lineHeight.small};
+`;
+
+const MODAL_WIDTH = '400px';
+
+export const STAR_RATING_FLOW = {
+    ONE: 'one',
+    TWO: 'two',
+    THREE: 'three',
+    FOUR: 'four',
+    FIVE: 'five',
+};
+
+const getStarRatingFlow = (data, star, step) => {
     const {
         OneStarRatingFlow,
         TwoStarRatingFlow,
         ThreeStarRatingFlow,
         FourStarRatingFlow,
         FiveStarRatingFlow,
     } = dlv(data, 'strapiFeedbackRatingFlow', {});
-    // Uncomment below to see shape of data, other forms
-    // console.log(data);
-    // TODO: Implement feedback UI
-    return (
-        <CMSForm form={OneStarRatingFlow ? OneStarRatingFlow.forms[1] : []} />
-    );
+
+    switch (star) {
+        case STAR_RATING_FLOW.ONE:
+            return {
+                ratingFlow: OneStarRatingFlow?.forms[step],
+                stepsCounter: OneStarRatingFlow?.forms?.length,
```

Comment:
Curious would `length` be `null` here if this value is undefined? If so that might cause an issue with the number comparison on line 109.

Also, would it be possible to generalize the return statement here into a function which we could pass in `OneStarRatingFlow?.forms` or such? That may help for maintenance in the future.

Reply:
If `stepsCounter` will be `undefined` or `null` comparison on line 109 will return false and render return nothing.